frontend/packages/ceph-storage-plugin/src/features.ts

```diff
@@ -20,19 +21,33 @@ export const detectIndependentMode: FeatureDetector = (dispatch) =>
     },
   );
 
-export const OCS_VERSION_4_5_FLAG = 'OCS_VERSION_4_5_FLAG';
-
-export const isOCS45AndAboveVersion = (subscription: K8sResourceKind): boolean => {
-  const version = subscription?.status?.currentCSV;
-  return version && version.includes('ocs-operator.v4') && version.split('.')[2] >= 5;
+export const OCS_SUPPORT_FLAGS = {
+  SNAPSHOT: 'SNAPSHOT',
+  // EXTERNAL: 'EXTERNAL',
 };
 
-export const detectOCSVersion45AndAbove: FeatureDetector = (dispatch) =>
-  k8sGet(SubscriptionModel, 'ocs-subscription', CEPH_STORAGE_NAMESPACE).then(
-    (obj: K8sResourceKind) => dispatch(setFlag(OCS_VERSION_4_5_FLAG, isOCS45AndAboveVersion(obj))),
-    (err) => {
-      err?.response?.status === 404
-        ? dispatch(setFlag(OCS_VERSION_4_5_FLAG, false))
-        : handleError(err, OCS_VERSION_4_5_FLAG, dispatch, detectOCSVersion45AndAbove);
+export const detectOCSSupportedFeatures: FeatureDetector = (dispatch) =>
+  // eslint-disable-next-line promise/catch-or-return
+  k8sGet(SubscriptionModel, 'ocs-operator', CEPH_STORAGE_NAMESPACE).then(
+    (subscription: K8sResourceKind) => {
+      // eslint-disable-next-line promise/catch-or-return
+      k8sGet(ClusterServiceVersionModel, subscription?.status?.currentCSV, CEPH_STORAGE_NAMESPACE) // eslint-disable-line promise/no-nesting
+        .then(
+          (obj: K8sResourceKind) => {
+            const support = obj?.metadata?.annotations['features.ocs.openshift.io/enabled'];
+            _.keys(OCS_SUPPORT_FLAGS).forEach((feature) => {
+              dispatch(setFlag(feature, support.includes(feature.toLowerCase())));
+            });
+          },
+          (err) => {
+            err?.response?.status === 404
+              ? _.keys(OCS_SUPPORT_FLAGS).forEach((feature) => {
+                  dispatch(setFlag(feature, false));
+                })
+              : _.keys(OCS_SUPPORT_FLAGS).forEach((feature) => {
+                  handleError(err, feature, dispatch, detectOCSSupportedFeatures);
```

Comment:
We can't reuse the `handleError` because it invokes a network call for every instance of the key. This is not the desired behaviour what we want is a custom handleError that will take in an array of flags instead of one flag at a time. I would suggest you define your own.